packages/go_router/tool/run_tests.dart

```diff
@@ -9,25 +9,103 @@
 // ignore_for_file: avoid_print
 
 import 'dart:io';
+import 'package:io/io.dart' as io;
 import 'package:path/path.dart' as p;
+import 'package:yaml_edit/yaml_edit.dart';
 
 Future<void> main(List<String> args) async {
-  if (!Platform.isMacOS) {
-    print('This test can only be run on macOS.');
+  if (!Platform.isMacOS && !Platform.isWindows) {
+    print('This test can only be run on macOS or windows.');
     exit(0);
   }
+  final p.Context ctx = p.context;
   final Directory packageRoot =
-      Directory(p.dirname(Platform.script.path)).parent;
+      Directory(ctx.dirname(_ensureTrimLeadingSeparator(Platform.script.path)))
+          .parent;
+
+  // Copy test_fixes/ to be tested in a temp directory.
+  // This ensures the dart fix can be applied to projects
+  // outside of this package.
+  final Directory testFixesTargetDir = await Directory.systemTemp.createTemp();
+
+  await _prepareTemplate(
+    ctx: ctx,
+    testFixesDir: ctx.join(packageRoot.path, 'test_fixes'),
+    testFixesTargetDir: testFixesTargetDir.path,
+  );
+  final int pubGet = await _runProcess(
+    'dart',
+    <String>[
+      'pub',
+      'upgrade',
+    ],
+    workingDirectory: testFixesTargetDir.path,
+  );
+  if (pubGet != 0) {
+    await cleanUpAndExit(statusCode: pubGet, toDelete: testFixesTargetDir);
+  }
   final int status = await _runProcess(
     'dart',
     <String>[
       'fix',
       '--compare-to-golden',
     ],
-    workingDirectory: p.join(packageRoot.path, 'test_fixes'),
+    workingDirectory: testFixesTargetDir.path,
   );
+  if (status != 0) {
+    await cleanUpAndExit(statusCode: status, toDelete: testFixesTargetDir);
+  }
+  await cleanUpAndExit(statusCode: 0, toDelete: testFixesTargetDir);
+}
 
-  exit(status);
+Future<Never> cleanUpAndExit({
+  required int statusCode,
+  required Directory toDelete,
+}) async {
+  await toDelete.delete(recursive: true);
+  exit(statusCode);
+}
+
+Future<void> _prepareTemplate({
+  required String testFixesDir,
+  required String testFixesTargetDir,
+  required p.Context ctx,
```

Comment:
Parameter name should not be abbreviated.

(Although per my earlier comment, it's not clear to me what purpose this is serving in the first place.)